hub/integrations/cleanlab/cleanlab.py

```diff
@@ -0,0 +1,157 @@
+            - label_issues: A boolean mask for the entire dataset where True represents a label issue and False represents an example that is confidently/accurately labeled.
+            - label_quality_scores: Label quality scores for each datapoint, where lower scores indicate labels less likely to be correct.
+            - predicted_labels: Class predicted by model trained on cleaned data for each example in the dataset.
+
+    Raises:
+        ...
+
+    """
+
+    from hub.integrations.cleanlab.label_issues import get_label_issues
+    from hub.integrations.common.utils import is_dataset
+
+    # Catch most common user errors early.
+    if not is_dataset(dataset):
+        raise TypeError(f"`dataset` must be a Hub Dataset. Got {type(dataset)}")
+
+    label_issues, label_quality_scores, predicted_labels = get_label_issues(
+        dataset=dataset,
+        model=model,
+        folds=folds,
+        pretrained=pretrained,
+        verbose=verbose,
+        label_issues_kwargs=label_issues_kwargs,
+        label_quality_kwargs=label_quality_kwargs,
+    )
+
+    return pd.DataFrame(
+        {
+            "is_label_issue": label_issues,
+            "label_quality": label_quality_scores,
+            "predicted_labels": predicted_labels,
+        }
+    )
+
+
+def create_tensors(
+    dataset: Any,
+    label_issues: Any,
+    branch: Union[str, None] = None,
+    overwrite: bool = False,
+    verbose: bool = True,
+):
+    """
+    Creates tensors `is_label_issue` and `label_quality_scores` and `predicted_labels` under `label_issues group`.
+
+    Note:
+        This method would only work if you have write access to the dataset.
+
+    Args:
+        dataset (class): Hub Dataset to add the tensors to.
+        label_issues (class): pandas DataFrame of label issues for each example computed by running `clean_labels()`.
+        branch (str, Optional): The name of the branch to use for creating the label_issues tensor group. If the branch name is provided but the branch does not exist, it will be created. If no branch is provided, the default branch will be used.
+        overwrite (bool): If True, will overwrite label_issues tensors if they already exists. Only applicable if `create_tensors` is True. Default is False.
+        verbose (bool): This parameter controls how much output is printed. Default is True.
+
+    Returns:
+        commit_id (str): The commit hash of the commit that was created.
+
+    Raises:
+        ...
+
+    """
+    from hub.integrations.cleanlab.tensors import create_label_issues_tensors
+    from hub.integrations.cleanlab.utils import switch_branch
+    from hub.integrations.common.utils import is_dataset
+
+    if not is_dataset(dataset):
+        raise TypeError(f"`dataset` must be a Hub Dataset. Got {type(dataset)}")
+
+    # Catch write access error early.
+    if dataset.read_only:
+        raise ValueError(
```

Comment:
Error message should be different if ds._locked_out is True. This means user wanted to open the dataset with write access but it's already locked for writing by someone else.